<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>References – AI in Microscopy: A BioImaging Guide</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./6-image-restoration-appendix.html" rel="next">
<link href="./glossary.html" rel="prev">
<link href="./settings/favicon.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark-707d8167ce6003fca903bfe2be84ab7f.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" class="quarto-color-scheme-extra" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-2641b481724464e61c86985d8c912b6f.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark-4c6d65c679321d81af7ec8b61b1d5a24.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="site_libs/bootstrap/bootstrap-2641b481724464e61c86985d8c912b6f.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme-extra" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-Y29EKZ8LWD"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-Y29EKZ8LWD', { 'anonymize_ip': true});
</script>


</head>

<body class="nav-sidebar floating quarto-light"><script id="quarto-html-before-body" type="application/javascript">
    const toggleBodyColorMode = (bsSheetEl) => {
      const mode = bsSheetEl.getAttribute("data-mode");
      const bodyEl = window.document.querySelector("body");
      if (mode === "dark") {
        bodyEl.classList.add("quarto-dark");
        bodyEl.classList.remove("quarto-light");
      } else {
        bodyEl.classList.add("quarto-light");
        bodyEl.classList.remove("quarto-dark");
      }
    }
    const toggleBodyColorPrimary = () => {
      const bsSheetEl = window.document.querySelector("link#quarto-bootstrap:not([rel=disabled-stylesheet])");
      if (bsSheetEl) {
        toggleBodyColorMode(bsSheetEl);
      }
    }
    const setColorSchemeToggle = (alternate) => {
      const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
      for (let i=0; i < toggles.length; i++) {
        const toggle = toggles[i];
        if (toggle) {
          if (alternate) {
            toggle.classList.add("alternate");
          } else {
            toggle.classList.remove("alternate");
          }
        }
      }
    };
    const toggleColorMode = (alternate) => {
      // Switch the stylesheets
      const primaryStylesheets = window.document.querySelectorAll('link.quarto-color-scheme:not(.quarto-color-alternate)');
      const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
      manageTransitions('#quarto-margin-sidebar .nav-link', false);
      if (alternate) {
        // note: dark is layered on light, we don't disable primary!
        enableStylesheet(alternateStylesheets);
        for (const sheetNode of alternateStylesheets) {
          if (sheetNode.id === "quarto-bootstrap") {
            toggleBodyColorMode(sheetNode);
          }
        }
      } else {
        disableStylesheet(alternateStylesheets);
        enableStylesheet(primaryStylesheets)
        toggleBodyColorPrimary();
      }
      manageTransitions('#quarto-margin-sidebar .nav-link', true);
      // Switch the toggles
      setColorSchemeToggle(alternate)
      // Hack to workaround the fact that safari doesn't
      // properly recolor the scrollbar when toggling (#1455)
      if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
        manageTransitions("body", false);
        window.scrollTo(0, 1);
        setTimeout(() => {
          window.scrollTo(0, 0);
          manageTransitions("body", true);
        }, 40);
      }
    }
    const disableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        stylesheet.rel = 'disabled-stylesheet';
      }
    }
    const enableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        if(stylesheet.rel !== 'stylesheet') { // for Chrome, which will still FOUC without this check
          stylesheet.rel = 'stylesheet';
        }
      }
    }
    const manageTransitions = (selector, allowTransitions) => {
      const els = window.document.querySelectorAll(selector);
      for (let i=0; i < els.length; i++) {
        const el = els[i];
        if (allowTransitions) {
          el.classList.remove('notransition');
        } else {
          el.classList.add('notransition');
        }
      }
    }
    const isFileUrl = () => {
      return window.location.protocol === 'file:';
    }
    const hasAlternateSentinel = () => {
      let styleSentinel = getColorSchemeSentinel();
      if (styleSentinel !== null) {
        return styleSentinel === "alternate";
      } else {
        return false;
      }
    }
    const setStyleSentinel = (alternate) => {
      const value = alternate ? "alternate" : "default";
      if (!isFileUrl()) {
        window.localStorage.setItem("quarto-color-scheme", value);
      } else {
        localAlternateSentinel = value;
      }
    }
    const getColorSchemeSentinel = () => {
      if (!isFileUrl()) {
        const storageValue = window.localStorage.getItem("quarto-color-scheme");
        return storageValue != null ? storageValue : localAlternateSentinel;
      } else {
        return localAlternateSentinel;
      }
    }
    const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
      const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
      const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
      let newTheme = '';
      if(authorPrefersDark) {
        newTheme = isAlternate ? baseTheme : alternateTheme;
      } else {
        newTheme = isAlternate ? alternateTheme : baseTheme;
      }
      const changeGiscusTheme = () => {
        // From: https://github.com/giscus/giscus/issues/336
        const sendMessage = (message) => {
          const iframe = document.querySelector('iframe.giscus-frame');
          if (!iframe) return;
          iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
        }
        sendMessage({
          setConfig: {
            theme: newTheme
          }
        });
      }
      const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
      if (isGiscussLoaded) {
        changeGiscusTheme();
      }
    };
    const authorPrefersDark = false;
    const darkModeDefault = authorPrefersDark;
      document.querySelector('link#quarto-text-highlighting-styles.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
      document.querySelector('link#quarto-bootstrap.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
    let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
    // Dark / light mode switch
    window.quartoToggleColorScheme = () => {
      // Read the current dark / light value
      let toAlternate = !hasAlternateSentinel();
      toggleColorMode(toAlternate);
      setStyleSentinel(toAlternate);
      toggleGiscusIfUsed(toAlternate, darkModeDefault);
      window.dispatchEvent(new Event('resize'));
    };
    // Switch to dark mode if need be
    if (hasAlternateSentinel()) {
      toggleColorMode(true);
    } else {
      toggleColorMode(false);
    }
  </script>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./references.html">References</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      </a>
    <div class="sidebar-title mb-0 py-0">
      <a href="./">AI in Microscopy: A BioImaging Guide</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/aicjanelia/BioImagingAI" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="./AI-in-Microscopy--A-BioImaging-Guide.pdf" title="Download PDF" class="quarto-navigation-tool px-1" aria-label="Download PDF"><i class="bi bi-file-pdf"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./1-intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Getting Started with AI</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./2-primer.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">AI Primer</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./3-llms.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Foundations of Large Language Models</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./4-architectures.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Architectures and Loss Models</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Image Acquisition</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./5-training-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Collecting Training Data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./6-image-restoration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Image Restoration</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./7-smart-microscopy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Adding AI to Your Hardware</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Image Analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./8-existing-tools.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Chapter 8: How do you select and find a tool?</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./9-train-models.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">How to Train and Use Deep Learning Models in Microscopy</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./10-output-quality.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Output Quality</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./11-outlook.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Outlook</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./glossary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Glossary</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">References</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">Appendices</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./6-image-restoration-appendix.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">A</span>&nbsp; <span class="chapter-title">3D-RCAN for Image Restoration</span></span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
        
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">


<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">References</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<div id="refs" class="references csl-bib-body" data-entry-spacing="0" data-line-spacing="2" role="list">
<div id="ref-stringer2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Stringer, C., Wang, T., Michaelos, M. &amp;
Pachitariu, M. <a href="https://doi.org/10.1038/s41592-020-01018-x">Cellpose: A generalist
algorithm for cellular segmentation</a>. <em>Nature Methods</em>
<strong>18</strong>, 100–106 (2021).</div>
</div>
<div id="ref-von_chamier2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Chamier, L. von <em>et al.</em> <a href="https://doi.org/10.1038/s41467-021-22518-0">Democratising deep
learning for microscopy with <span>ZeroCostDL4Mic</span></a>. <em>Nature
Communications</em> <strong>12</strong>, 2276 (2021).</div>
</div>
<div id="ref-guo_deep_2025" class="csl-entry" role="listitem">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Guo,
M. <em>et al.</em> <a href="https://doi.org/10.1038/s41467-024-55267-x">Deep learning-based
aberration compensation improves contrast and resolution in fluorescence
microscopy</a>. <em>Nature Communications</em> <strong>16</strong>, 313
(2025).</div>
</div>
<div id="ref-Wu2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Wu,
Y. &amp; Shroff, H. <a href="https://doi.org/10.1007/s00418-022-02147-4">Multiscale
fluorescence imaging of living samples</a>. <em>Histochemistry and Cell
Biology</em> <strong>158</strong>, 301–323 (2022).</div>
</div>
<div id="ref-Schermelleh2010" class="csl-entry" role="listitem">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Schermelleh, L., Heintzmann, R. &amp;
Leonhardt, H. <a href="https://doi.org/10.1083/jcb.201002018">A guide to
super-resolution fluorescence microscopy</a>. <em>Journal of Cell
Biology</em> <strong>190</strong>, 165–175 (2010).</div>
</div>
<div id="ref-Archit2025" class="csl-entry" role="listitem">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Archit, A. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-024-02580-4">Segment anything for
microscopy</a>. <em>Nature Methods</em> <strong>22</strong>, 579–591
(2025).</div>
</div>
<div id="ref-Sahl2017" class="csl-entry" role="listitem">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">Sahl, S. J., Hell, S. W. &amp; Jakobs, S. <a href="https://doi.org/10.1038/nrm.2017.71">Fluorescence nanoscopy in
cell biology</a>. <em>Nature Reviews Molecular Cell Biology</em>
<strong>18</strong>, 685–701 (2017).</div>
</div>
<div id="ref-Schermelleh2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">Schermelleh, L. <em>et al.</em> <a href="https://doi.org/10.1038/s41556-018-0251-8">Super-resolution
microscopy demystified</a>. <em>Nature Cell Biology</em>
<strong>21</strong>, 72–84 (2019).</div>
</div>
<div id="ref-Ji2017" class="csl-entry" role="listitem">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">Ji,
N. <a href="https://doi.org/10.1038/nmeth.4218">Adaptive optical
fluorescence microscopy</a>. <em>Nature Methods</em>
<strong>14</strong>, 374–380 (2017).</div>
</div>
<div id="ref-Hampson2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">Hampson, K. M. <em>et al.</em> <a href="https://doi.org/10.1038/s43586-021-00066-7">Adaptive optics for
high-resolution imaging</a>. <em>Nature Reviews Methods Primers</em>
<strong>1</strong>, 68 (2021).</div>
</div>
<div id="ref-Shroff2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">11. </div><div class="csl-right-inline">Shroff, H., Testa, I., Jug, F. &amp; Manley, S.
<a href="https://doi.org/10.1038/s41580-024-00702-6">Live-cell imaging
powered by computation</a>. <em>Nature Reviews Molecular Cell
Biology</em> <strong>25</strong>, 443–463 (2024).</div>
</div>
<div id="ref-Venkatesh2015" class="csl-entry" role="listitem">
<div class="csl-left-margin">12. </div><div class="csl-right-inline">Venkatesh, M., Mohan, K. &amp; Seelamantula, C.
S. <a href="https://doi.org/10.1063/1.4930029">Directional bilateral
filters for smoothing fluorescence microscopy images</a>. <em>AIP
Advances</em> <strong>5</strong>, 084805 (2015).</div>
</div>
<div id="ref-Danielyan2014" class="csl-entry" role="listitem">
<div class="csl-left-margin">13. </div><div class="csl-right-inline">Danielyan, A., Wu, Y.-W., Shih, P.-Y.,
Dembitskaya, Y. &amp; Semyanov, A. <a href="https://doi.org/10.1016/j.ymeth.2014.03.010">Denoising of
two-photon fluorescence images with block-matching 3D filtering</a>.
<em>Methods</em> <strong>68</strong>, 308–316 (2014).</div>
</div>
<div id="ref-Zhang2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">14. </div><div class="csl-right-inline">Zhang, Y. <em>et al.</em> A poisson-gaussian
denoising dataset with real fluorescence microscopy images. in <em>2019
IEEE/CVF conference on computer vision and pattern recognition
(CVPR)</em> 11702–11710 (Optica Publishing Group, 2019). doi:<a href="https://doi.org/10.1109/CVPR.2019.01198">10.1109/CVPR.2019.01198</a>.</div>
</div>
<div id="ref-Li2017" class="csl-entry" role="listitem">
<div class="csl-left-margin">15. </div><div class="csl-right-inline">Li,
J., Luisier, F. &amp; Blu, T. Pure-let deconvolution of 3D fluorescence
microscopy images. in <em>2017 IEEE 14th international symposium on
biomedical imaging (ISBI 2017)</em> 723–727 (2017). doi:<a href="https://doi.org/10.1109/ISBI.2017.7950621">10.1109/ISBI.2017.7950621</a>.</div>
</div>
<div id="ref-Makitalo2013" class="csl-entry" role="listitem">
<div class="csl-left-margin">16. </div><div class="csl-right-inline">Makitalo, M. &amp; Foi, A. <a href="https://doi.org/10.1109/TIP.2012.2202675">Optimal inversion of the
generalized anscombe transformation for poisson-gaussian noise</a>.
<em>IEEE Transactions on Image Processing</em> <strong>22</strong>,
91–103 (2013).</div>
</div>
<div id="ref-Luisier2010" class="csl-entry" role="listitem">
<div class="csl-left-margin">17. </div><div class="csl-right-inline">Luisier, F., Vonesch, C., Blu, T. &amp; Unser,
M. <a href="https://doi.org/10.1016/j.sigpro.2009.07.009">Fast
interscale wavelet denoising of poisson-corrupted images</a>. <em>Signal
Processing</em> <strong>90</strong>, 415–427 (2010).</div>
</div>
<div id="ref-Van_der_Walt2014" class="csl-entry" role="listitem">
<div class="csl-left-margin">18. </div><div class="csl-right-inline">Walt, S. van der <em>et al.</em> <a href="https://peerj.com/articles/453/">Scikit-image: Image processing in
python</a>. <em>PeerJ</em> <strong>2</strong>, e453 (2014).</div>
</div>
<div id="ref-Wiener1949" class="csl-entry" role="listitem">
<div class="csl-left-margin">19. </div><div class="csl-right-inline">Wiener, N. <em>Extrapolation, Interpolation,
and Smoothing of Stationary Time Series: With Engineering
Applications</em>. (The MIT Press, 1949). doi:<a href="https://doi.org/10.7551/mitpress/2946.001.0001">10.7551/mitpress/2946.001.0001</a>.</div>
</div>
<div id="ref-Tikhonov1963" class="csl-entry" role="listitem">
<div class="csl-left-margin">20. </div><div class="csl-right-inline">Tikhonov, A. N. Solution of incorrectly
formulated problems and the regularization method. <em>Soviet Math.
Dokl.</em> <strong>4</strong>, 1035–1038 (1963).</div>
</div>
<div id="ref-Miller1970" class="csl-entry" role="listitem">
<div class="csl-left-margin">21. </div><div class="csl-right-inline">Miller, K. <a href="https://doi.org/10.1137/0501006">Least squares methods for
ill-posed problems with a prescribed bound</a>. <em>SIAM Journal on
Mathematical Analysis</em> <strong>1</strong>, 52–74 (1970).</div>
</div>
<div id="ref-Beck2009" class="csl-entry" role="listitem">
<div class="csl-left-margin">22. </div><div class="csl-right-inline">Beck, A. &amp; Teboulle, M. <a href="https://doi.org/10.1137/080716542">A fast iterative
shrinkage-thresholding algorithm for linear inverse problems</a>.
<em>SIAM Journal on Imaging Sciences</em> <strong>2</strong>, 183–202
(2009).</div>
</div>
<div id="ref-Lucy1974" class="csl-entry" role="listitem">
<div class="csl-left-margin">23. </div><div class="csl-right-inline">Lucy, L. B. <a href="https://doi.org/10.1086/111605"><span class="nocase">An iterative
technique for the rectification of observed distributions</span></a>.
<em>Astronomical Journal</em> <strong>79</strong>, 745 (1974).</div>
</div>
<div id="ref-Richardson1972" class="csl-entry" role="listitem">
<div class="csl-left-margin">24. </div><div class="csl-right-inline">Richardson, W. H. <a href="https://doi.org/10.1364/JOSA.62.000055">Bayesian-based iterative
method of image restoration<span class="math inline">*</span></a>.
<em>J. Opt. Soc. Am.</em> <strong>62</strong>, 55–59 (1972).</div>
</div>
<div id="ref-Sarder2006" class="csl-entry" role="listitem">
<div class="csl-left-margin">25. </div><div class="csl-right-inline">Sarder, P. &amp; Nehorai, A. <a href="https://doi.org/10.1109/MSP.2006.1628876">Deconvolution methods
for 3-d fluorescence microscopy images</a>. <em>IEEE Signal Processing
Magazine</em> <strong>23</strong>, 32–45 (2006).</div>
</div>
<div id="ref-Goodwin2014" class="csl-entry" role="listitem">
<div class="csl-left-margin">26. </div><div class="csl-right-inline">Goodwin, P. C. <a href="https://doi.org/10.1016/B978-0-12-420138-5.00010-0">Chapter 10 -
quantitative deconvolution microscopy</a>. in <em>Quantitative imaging
in cell biology</em> (eds. Waters, J. C. &amp; Wittman, T.) vol. 123
177–192 (Academic Press, 2014).</div>
</div>
<div id="ref-Guo2020" class="csl-entry" role="listitem">
<div class="csl-left-margin">27. </div><div class="csl-right-inline">Guo, M. <em>et al.</em> <a href="https://doi.org/10.1038/s41587-020-0560-x">Rapid image
deconvolution and multiview fusion for optical microscopy</a>.
<em>Nature Biotechnology</em> <strong>38</strong>, 1337–1346
(2020).</div>
</div>
<div id="ref-Schindelin2012" class="csl-entry" role="listitem">
<div class="csl-left-margin">28. </div><div class="csl-right-inline">Schindelin, J. <em>et al.</em> <a href="https://doi.org/10.1038/nmeth.2019">Fiji: An open-source platform
for biological-image analysis</a>. <em>Nature Methods</em>
<strong>9</strong>, 676–682 (2012).</div>
</div>
<div id="ref-Sage2017" class="csl-entry" role="listitem">
<div class="csl-left-margin">29. </div><div class="csl-right-inline">Sage, D. <em>et al.</em> <a href="https://doi.org/10.1016/j.ymeth.2016.12.015">DeconvolutionLab2: An
open-source software for deconvolution microscopy</a>. <em>Methods</em>
<strong>115</strong>, 28–41 (2017).</div>
</div>
<div id="ref-Bazin2007" class="csl-entry" role="listitem">
<div class="csl-left-margin">30. </div><div class="csl-right-inline">Bazin, P.-L. <em>et al.</em> <a href="https://doi.org/10.1016/j.jneumeth.2007.05.024">Volumetric
neuroimage analysis extensions for the MIPAV software package</a>.
<em>Journal of Neuroscience Methods</em> <strong>165</strong>, 111–121
(2007).</div>
</div>
<div id="ref-Booth1861" class="csl-entry" role="listitem">
<div class="csl-left-margin">31. </div><div class="csl-right-inline">Booth, M. J. <a href="http://www.jstor.org/stable/25190627">Adaptive optics in
microscopy</a>. <em>Philosophical Transactions: Mathematical, Physical
and Engineering Sciences</em> <strong>365</strong>, 2829–2843
(2007).</div>
</div>
<div id="ref-Hell2007" class="csl-entry" role="listitem">
<div class="csl-left-margin">32. </div><div class="csl-right-inline">Hell, S. W. <a href="https://doi.org/10.1126/science.1137395">Far-field optical
nanoscopy</a>. <em>Science</em> <strong>316</strong>, 1153–1158
(2007).</div>
</div>
<div id="ref-Vicidomini2018" class="csl-entry" role="listitem">
<div class="csl-left-margin">33. </div><div class="csl-right-inline">Vicidomini, G., Bianchini, P. &amp; Diaspro, A.
<a href="https://doi.org/10.1038/nmeth.4593">STED super-resolved
microscopy</a>. <em>Nature Methods</em> <strong>15</strong>, 173–182
(2018).</div>
</div>
<div id="ref-Wu2018" class="csl-entry" role="listitem">
<div class="csl-left-margin">34. </div><div class="csl-right-inline">Wu,
Y. &amp; Shroff, H. <a href="https://doi.org/10.1038/s41592-018-0211-z">Faster, sharper, and
deeper: Structured illumination microscopy for biological imaging</a>.
<em>Nature Methods</em> <strong>15</strong>, 1011–1019 (2018).</div>
</div>
<div id="ref-Chen2015" class="csl-entry" role="listitem">
<div class="csl-left-margin">35. </div><div class="csl-right-inline">Chen, F., Tillberg, P. W. &amp; Boyden, E. S.
<a href="https://doi.org/10.1126/science.1260088">Expansion
microscopy</a>. <em>Science</em> <strong>347</strong>, 543–548
(2015).</div>
</div>
<div id="ref-Wassie2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">36. </div><div class="csl-right-inline">Wassie, A. T., Zhao, Y. &amp; Boyden, E. S. <a href="https://doi.org/10.1038/s41592-018-0219-4">Expansion microscopy:
Principles and uses in biological research</a>. <em>Nature Methods</em>
vol. 16 33–41 (2019).</div>
</div>
<div id="ref-Valli2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">37. </div><div class="csl-right-inline">Valli, J. <em>et al.</em> <a href="https://doi.org/10.1016/j.jbc.2021.100791">Seeing beyond the
limit: A guide to choosing the right super-resolution microscopy
technique</a>. <em>Journal of Biological Chemistry</em>
<strong>297</strong>, 100791 (2021).</div>
</div>
<div id="ref-Chen2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">38. </div><div class="csl-right-inline">Chen, H. <em>et al.</em> <a href="https://doi.org/10.1021/cbmi.4c00019">Advancements and practical
considerations for biophysical research: Navigating the challenges and
future of super-resolution microscopy</a>. <em>Chemical
<span>&amp;</span> Biomedical Imaging</em> <strong>2</strong>, 331–344
(2024).</div>
</div>
<div id="ref-Hagen2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">39. </div><div class="csl-right-inline">Hagen, G. M. <em>et al.</em> <a href="https://doi.org/10.1093/gigascience/giab032">Fluorescence
microscopy datasets for training deep neural networks</a>.
<em>GigaScience</em> <strong>10</strong>, giab032 (2021).</div>
</div>
<div id="ref-Weigert2018" class="csl-entry" role="listitem">
<div class="csl-left-margin">40. </div><div class="csl-right-inline">Weigert, M. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-018-0216-7">Content-aware image
restoration: Pushing the limits of fluorescence microscopy</a>.
<em>Nature Methods</em> <strong>15</strong>, 1090–1097 (2018).</div>
</div>
<div id="ref-Chen2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">41. </div><div class="csl-right-inline">Chen, J. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-021-01155-x">Three-dimensional
residual channel attention networks denoise and sharpen fluorescence
microscopy image volumes</a>. <em>Nature Methods</em>
<strong>18</strong>, 678–687 (2021).</div>
</div>
<div id="ref-Qiao2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">42. </div><div class="csl-right-inline">Qiao, C. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-020-01048-5">Evaluation and
development of deep neural networks for image super-resolution in
optical microscopy</a>. <em>Nature Methods</em> <strong>18</strong>,
194–202 (2021).</div>
</div>
<div id="ref-Hou2025" class="csl-entry" role="listitem">
<div class="csl-left-margin">43. </div><div class="csl-right-inline">Hou, X. <em>et al.</em> <a href="https://doi.org/10.1364/OE.554927">HD2Net: A deep learning
framework for simultaneous denoising and deaberration in fluorescence
microscopy</a>. <em>Opt. Express</em> <strong>33</strong>, 27317–27333
(2025).</div>
</div>
<div id="ref-Zhang2017" class="csl-entry" role="listitem">
<div class="csl-left-margin">44. </div><div class="csl-right-inline">Zhang, K., Zuo, W., Chen, Y., Meng, D. &amp;
Zhang, L. <a href="https://doi.org/10.1109/TIP.2017.2662206">Beyond a
gaussian denoiser: Residual learning of deep CNN for image
denoising</a>. <em>IEEE Transactions on Image Processing</em>
<strong>26</strong>, 3142–3155 (2017).</div>
</div>
<div id="ref-Dabov2007" class="csl-entry" role="listitem">
<div class="csl-left-margin">45. </div><div class="csl-right-inline">Dabov, K., Foi, A., Katkovnik, V. &amp;
Egiazarian, K. <a href="https://doi.org/10.1109/TIP.2007.901238">Image
denoising by sparse 3-d transform-domain collaborative filtering</a>.
<em>IEEE Transactions on Image Processing</em> <strong>16</strong>,
2080–2095 (2007).</div>
</div>
<div id="ref-Krull2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">46. </div><div class="csl-right-inline">Krull, A., Buchholz, T.-O. &amp; Jug, F.
Noise2Void - learning denoising from single noisy images. in <em>2019
IEEE/CVF conference on computer vision and pattern recognition
(CVPR)</em> 2124–2132 (2019). doi:<a href="https://doi.org/10.1109/CVPR.2019.00223">10.1109/CVPR.2019.00223</a>.</div>
</div>
<div id="ref-Batson2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">47. </div><div class="csl-right-inline">Batson, J. &amp; Royer, L. <a href="https://proceedings.mlr.press/v97/batson19a.html"><span>N</span>oise2<span>S</span>elf:
Blind denoising by self-supervision</a>. in <em>Proceedings of the 36th
international conference on machine learning</em> (eds. Chaudhuri, K.
&amp; Salakhutdinov, R.) vol. 97 524–533 (PMLR, 2019).</div>
</div>
<div id="ref-Lehtinen2018" class="csl-entry" role="listitem">
<div class="csl-left-margin">48. </div><div class="csl-right-inline">Lehtinen, J. <em>et al.</em> <a href="https://proceedings.mlr.press/v80/lehtinen18a.html"><span>N</span>oise2<span>N</span>oise:
Learning image restoration without clean data</a>. in <em>Proceedings of
the 35th international conference on machine learning</em> (eds. Dy, J.
&amp; Krause, A.) vol. 80 2965–2974 (PMLR, 2018).</div>
</div>
<div id="ref-Li2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">49. </div><div class="csl-right-inline">Li,
Y. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-022-01652-7">Incorporating the
image formation process into deep learning improves network
performance</a>. <em>Nature Methods</em> <strong>19</strong>, 1427–1437
(2022).</div>
</div>
<div id="ref-Yanny2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">50. </div><div class="csl-right-inline">Yanny, K., Monakhova, K., Shuai, R. W. &amp;
Waller, L. <a href="https://doi.org/10.1364/OPTICA.442438">Deep learning
for fast spatially varying deconvolution</a>. <em>Optica</em>
<strong>9</strong>, 96–99 (2022).</div>
</div>
<div id="ref-Saha2020" class="csl-entry" role="listitem">
<div class="csl-left-margin">51. </div><div class="csl-right-inline">Saha, D. <em>et al.</em> <a href="https://doi.org/10.1364/OE.401933">Practical sensorless aberration
estimation for 3D microscopy with deep learning</a>. <em>Opt.
Express</em> <strong>28</strong>, 29044–29053 (2020).</div>
</div>
<div id="ref-Kang2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">52. </div><div class="csl-right-inline">Kang, I., Zhang, Q., Yu, S. X. &amp; Ji, N. <a href="https://doi.org/10.1038/s42256-024-00853-3">Coordinate-based
neural representations for computational adaptive optics in widefield
microscopy</a>. <em>Nature Machine Intelligence</em> <strong>6</strong>,
714–725 (2024).</div>
</div>
<div id="ref-Kang2024.10.20.619284" class="csl-entry" role="listitem">
<div class="csl-left-margin">53. </div><div class="csl-right-inline">Kang, I. <em>et al.</em> Adaptive optical
correction in in vivo two-photon fluorescence microscopy with neural
fields. <em>bioRxiv</em> (2024) doi:<a href="https://doi.org/10.1101/2024.10.20.619284">10.1101/2024.10.20.619284</a>.</div>
</div>
<div id="ref-Fersini2025" class="csl-entry" role="listitem">
<div class="csl-left-margin">54. </div><div class="csl-right-inline">Fersini, F. <em>et al.</em> <a href="https://doi.org/10.1364/BOE.559899">Wavefront estimation through
structured detection in laser scanning microscopy</a>. <em>Biomed. Opt.
Express</em> <strong>16</strong>, 2135–2155 (2025).</div>
</div>
<div id="ref-Zhou2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">55. </div><div class="csl-right-inline">Zhou, Y., Jin, Z., Zhao, Q., Xiong, B. &amp;
Cao, X. <a href="https://doi.org/10.1002/lpor.202300154">Aberration
modeling in deep learning for volumetric reconstruction of light-field
microscopy</a>. <em>Laser &amp; Photonics Reviews</em>
<strong>17</strong>, 2300154 (2023).</div>
</div>
<div id="ref-Qiao2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">56. </div><div class="csl-right-inline">Qiao, C. <em>et al.</em> <a href="https://doi.org/10.1364/PRJ.506778">Deep learning-based optical
aberration estimation enables offline digital adaptive optics and
super-resolution imaging</a>. <em>Photon. Res.</em> <strong>12</strong>,
474–484 (2024).</div>
</div>
<div id="ref-Hu2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">57. </div><div class="csl-right-inline">Hu,
L., Hu, S., Gong, W. &amp; Si, K. <a href="https://doi.org/10.1364/OL.418997">Image enhancement for
fluorescence microscopy based on deep learning with prior knowledge of
aberration</a>. <em>Opt. Lett.</em> <strong>46</strong>, 2055–2058
(2021).</div>
</div>
<div id="ref-Wang2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">58. </div><div class="csl-right-inline">Wang, H. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-018-0239-0">Deep learning enables
cross-modality super-resolution in fluorescence microscopy</a>.
<em>Nature Methods</em> <strong>16</strong>, 103–110 (2019).</div>
</div>
<div id="ref-Park2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">59. </div><div class="csl-right-inline">Park, H. <em>et al.</em> <a href="https://doi.org/10.1038/s41467-022-30949-6">Deep learning enables
reference-free isotropic super-resolution for volumetric fluorescence
microscopy</a>. <em>Nature Communications</em> <strong>13</strong>, 3297
(2022).</div>
</div>
<div id="ref-Ning2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">60. </div><div class="csl-right-inline">Ning, K. <em>et al.</em> <a href="https://doi.org/10.1038/s41377-023-01230-2">Deep self-learning
enables fast, high-fidelity isotropic resolution restoration for
volumetric fluorescence microscopy</a>. <em>Light: Science
<span>&amp;</span> Applications</em> <strong>12</strong>, 204
(2023).</div>
</div>
<div id="ref-Ronneberger2015" class="csl-entry" role="listitem">
<div class="csl-left-margin">61. </div><div class="csl-right-inline">Ronneberger, O., Fischer, P. &amp; Brox, T.
U-net: Convolutional networks for biomedical image segmentation. in
<em>Medical image computing and computer-assisted intervention – MICCAI
2015</em> (eds. Navab, N., Hornegger, J., Wells, W. M. &amp; Frangi, A.
F.) 234–241 (Springer International Publishing, Cham, 2015). doi:<a href="https://doi.org/10.1007/978-3-319-24574-4_28">10.1007/978-3-319-24574-4_28</a>.</div>
</div>
<div id="ref-Ji2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">62. </div><div class="csl-right-inline">Ji,
Z., Li, J. D. &amp; Telgarsky, M. <a href="https://dl.acm.org/doi/10.5555/3540261.3540400">Early-stopped
neural networks are consistent</a>. in <em>Advances in neural
information processing systems 34 - 35th conference on neural
information processing systems, NeurIPS 2021</em> (eds. Ranzato, M.,
Beygelzimer, A., Dauphin, Y., Liang, {Percy. S. }. &amp; Vaughan}, J.
{Wortman) 1805–1817 (Neural information processing systems foundation,
2021).</div>
</div>
<div id="ref-Santos2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">63. </div><div class="csl-right-inline">Santos, C. F. G. D. &amp; Papa, J. P. <a href="https://doi.org/10.1145/3510413">Avoiding overfitting: A survey on
regularization methods for convolutional neural networks</a>. <em>ACM
Comput. Surv.</em> <strong>54</strong>, (2022).</div>
</div>
<div id="ref-Miseta2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">64. </div><div class="csl-right-inline">Miseta, T., Fodor, A. &amp; Vathy-Fogarassy, Á.
<a href="https://doi.org/10.1016/j.neucom.2023.127028">Surpassing early
stopping: A novel correlation-based stopping criterion for neural
networks</a>. <em>Neurocomputing</em> <strong>567</strong>, 127028
(2024).</div>
</div>
<div id="ref-Shah2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">65. </div><div class="csl-right-inline">Shah, Z. H. <em>et al.</em> <a href="https://doi.org/10.3389/frai.2024.1353873">Image restoration in
frequency space using complex-valued CNNs</a>. <em>Frontiers in
Artificial Intelligence</em> <strong>Volume 7 - 2024</strong>,
(2024).</div>
</div>
<div id="ref-Liu2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">66. </div><div class="csl-right-inline">Liu, J., Gao, F., Zhang, L. &amp; Yang, H. <a href="https://doi.org/10.3390/mi15070928">A saturation artifacts
inpainting method based on two-stage GAN for fluorescence microscope
images</a>. <em>Micromachines</em> <strong>15</strong>, (2024).</div>
</div>
<div id="ref-Bouchard2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">67. </div><div class="csl-right-inline">Bouchard, C. <em>et al.</em> <a href="https://doi.org/10.1038/s42256-023-00689-3">Resolution enhancement
with a task-assisted GAN to guide optical nanoscopy image analysis and
acquisition</a>. <em>Nature Machine Intelligence</em>
<strong>5</strong>, 830–844 (2023).</div>
</div>
<div id="ref-Qiao2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">68. </div><div class="csl-right-inline">Qiao, C. <em>et al.</em> <a href="https://doi.org/10.1038/s41587-022-01471-3">Rationalized deep
learning super-resolution microscopy for sustained live imaging of rapid
subcellular processes</a>. <em>Nature Biotechnology</em>
<strong>41</strong>, 367–377 (2023).</div>
</div>
<div id="ref-Zhong2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">69. </div><div class="csl-right-inline">Zhong, L., Liu, G. &amp; Yang, G. <a href="https://doi.org/10.1109/ISBI48211.2021.9434150">Blind denoising of
fluorescence microscopy images using GAN-based global noise
modeling</a>. in <em>ISBI</em> 863–867 (2021).</div>
</div>
<div id="ref-Park2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">70. </div><div class="csl-right-inline">Park, E. <em>et al.</em> <a href="https://doi.org/10.1038/s41467-024-55262-2">Unsupervised
inter-domain transformation for virtually stained high-resolution
mid-infrared photoacoustic microscopy using explainable deep
learning</a>. <em>Nature Communications</em> <strong>15</strong>, 10892
(2024).</div>
</div>
<div id="ref-Osuna-Vargas2025" class="csl-entry" role="listitem">
<div class="csl-left-margin">71. </div><div class="csl-right-inline">Osuna-Vargas, P. <em>et al.</em> Denoising
diffusion models for high-resolution microscopy image restoration. in
<em>2025 IEEE/CVF winter conference on applications of computer vision
(WACV)</em> 4320–4330 (2025). doi:<a href="https://doi.org/10.1109/WACV61041.2025.00424">10.1109/WACV61041.2025.00424</a>.</div>
</div>
<div id="ref-Zhang2018" class="csl-entry" role="listitem">
<div class="csl-left-margin">72. </div><div class="csl-right-inline">Zhang, Y. <em>et al.</em> <a href="https://arxiv.org/abs/1807.02758">Image super-resolution using
very deep residual channel attention networks</a>. (2018).</div>
</div>
<div id="ref-Wang2004" class="csl-entry" role="listitem">
<div class="csl-left-margin">73. </div><div class="csl-right-inline">Wang, Z., Bovik, A. C., Sheikh, H. R. &amp;
Simoncelli, E. P. <a href="https://doi.org/10.1109/TIP.2003.819861">Image quality assessment:
From error visibility to structural similarity</a>. <em>IEEE
Transactions on Image Processing</em> <strong>13</strong>, 600–612
(2004).</div>
</div>
<div id="ref-Gonzalez2008" class="csl-entry" role="listitem">
<div class="csl-left-margin">74. </div><div class="csl-right-inline">Gonzalez, R. C. &amp; Woods, R. E. <em><a href="https://books.google.com/books?id=8uGOnjRGEzoC">Digital Image
Processing</a></em>. (Prentice Hall, 2008).</div>
</div>
<div id="ref-Haase2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">75. </div><div class="csl-right-inline">Haase, R., Tischer, C., Bankhead, P., Miura, K.
&amp; Cimini, B. <a href="https://osf.io/preprints/osf/2zgmc_v1">A call
for <span>FAIR</span> and open-access training materials to advance
<span>BioImage</span> analysis</a>. (2024).</div>
</div>
<div id="ref-Ljosa2012" class="csl-entry" role="listitem">
<div class="csl-left-margin">76. </div><div class="csl-right-inline">Ljosa, V., Sokolnicki, K. L. &amp; Carpenter,
A. E. <a href="https://doi.org/10.1038/nmeth.2083">Annotated
high-throughput microscopy image sets for validation</a>. <em>Nat.
Methods</em> <strong>9</strong>, 637 (2012).</div>
</div>
<div id="ref-Cimini2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">77. </div><div class="csl-right-inline">Cimini, B. A. <a href="https://carpenter-singh-lab.broadinstitute.org/blog/when-to-say-good-enough">When
to say ’good enough’</a>. (2019).</div>
</div>
<div id="ref-Jamali2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">78. </div><div class="csl-right-inline">Jamali, N., Dobson, E. T. A., Eliceiri, K. W.,
Carpenter, A. E. &amp; Cimini, B. A. <a href="https://doi.org/10.1017/S2633903X21000039">2020
<span>BioImage</span> analysis survey: Community experiences and needs
for the future</a>. <em>Biological Imaging</em> <strong>1</strong>, e4
(2021).</div>
</div>
<div id="ref-Sivagurunathan2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">79. </div><div class="csl-right-inline">Sivagurunathan, S. <em>et al.</em> <a href="https://doi.org/10.1111/jmi.13229">Bridging imaging users to
imaging analysis - a community survey</a>. <em>J. Microsc.</em>
(2023).</div>
</div>
<div id="ref-Schindelin2015" class="csl-entry" role="listitem">
<div class="csl-left-margin">80. </div><div class="csl-right-inline">Schindelin, J., Rueden, C. T., Hiner, M. C.
&amp; Eliceiri, K. W. <a href="https://doi.org/10.1002/mrd.22489">The
<span>ImageJ</span> ecosystem: An open platform for biomedical image
analysis</a>. <em>Mol. Reprod. Dev.</em> <strong>82</strong>, 518–529
(2015).</div>
</div>
<div id="ref-napari" class="csl-entry" role="listitem">
<div class="csl-left-margin">81. </div><div class="csl-right-inline">Napari: A multi-dimensional image viewer for
python. doi:<a href="https://doi.org/10.5281/zenodo.3555620">10.5281/zenodo.3555620</a>.</div>
</div>
<div id="ref-Stirling2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">82. </div><div class="csl-right-inline">Stirling, D. R. <em>et al.</em> <a href="https://doi.org/10.1186/s12859-021-04344-9"><span>CellProfiler</span>
4: Improvements in speed, utility and usability</a>. <em>BMC
Bioinformatics</em> <strong>22</strong>, 433 (2021).</div>
</div>
<div id="ref-Bankhead2017" class="csl-entry" role="listitem">
<div class="csl-left-margin">83. </div><div class="csl-right-inline">Bankhead, P. <em>et al.</em> <a href="https://doi.org/10.1038/s41598-017-17204-5"><span>QuPath</span>:
Open source software for digital pathology image analysis</a>. <em>Sci.
Rep.</em> <strong>7</strong>, 16878 (2017).</div>
</div>
<div id="ref-de-Chaumont2012" class="csl-entry" role="listitem">
<div class="csl-left-margin">84. </div><div class="csl-right-inline">Chaumont, F. de <em>et al.</em> <a href="https://doi.org/10.1038/nmeth.2075">Icy: An open bioimage
informatics platform for extended reproducible research</a>. <em>Nat.
Methods</em> <strong>9</strong>, 690–696 (2012).</div>
</div>
<div id="ref-Berg2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">85. </div><div class="csl-right-inline">Berg, S. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-019-0582-9">Ilastik: Interactive
machine learning for (bio)image analysis</a>. <em>Nat. Methods</em>
<strong>16</strong>, 1226–1232 (2019).</div>
</div>
<div id="ref-Arzt2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">86. </div><div class="csl-right-inline">Arzt, M. <em>et al.</em> <a href="https://www.frontiersin.org/articles/10.3389/fcomp.2022.777728"><span>LABKIT</span>:
Labeling and segmentation toolkit for big image data</a>. <em>Front.
Comput. Sci.</em> <strong>4</strong>, (2022).</div>
</div>
<div id="ref-Schmidt2018" class="csl-entry" role="listitem">
<div class="csl-left-margin">87. </div><div class="csl-right-inline">Schmidt, U., Weigert, M., Broaddus, C. &amp;
Myers, G. Cell detection with star-convex polygons. in <em>Medical image
computing and computer assisted intervention – MICCAI 2018</em> (eds.
Frangi, A. F., Schnabel, J. A., Davatzikos, C., Alberola-López, C. &amp;
Fichtinger, G.) 265–273 (Springer International Publishing, Cham, 2018).
doi:<a href="https://doi.org/10.1007/978-3-030-00934-2_30">10.1007/978-3-030-00934-2_30</a>.</div>
</div>
<div id="ref-Goldsborough2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">88. </div><div class="csl-right-inline">Goldsborough, T. <em>et al.</em> <a href="https://doi.org/10.48550/arXiv.2408.15954"><span>InstanSeg</span>:
An embedding-based instance segmentation algorithm optimized for
accurate, efficient and portable cell segmentation</a>. <em>arXiv
[cs.CV]</em> (2024).</div>
</div>
<div id="ref-Pachitariu2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">89. </div><div class="csl-right-inline">Pachitariu, M. &amp; Stringer, C. <a href="https://doi.org/10.1038/s41592-022-01663-4">Cellpose 2.0: How to
train your own model</a>. <em>Nat. Methods</em> <strong>19</strong>,
1634–1641 (2022).</div>
</div>
<div id="ref-Zhang2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">90. </div><div class="csl-right-inline">Zhang, C. <em>et al.</em> <a href="https://doi.org/10.48550/arXiv.2312.11256">Bio-image informatics
index <span>BIII</span>: A unique database of image analysis tools and
workflows for and by the bioimaging community</a>. <em>arXiv
[q-bio.QM]</em> (2023).</div>
</div>
<div id="ref-Ouyang2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">91. </div><div class="csl-right-inline">Ouyang, W. <em>et al.</em>
<span>BioImage</span> model zoo: A community-driven resource for
accessible deep learning in <span>BioImage</span> analysis.
<em>bioRxiv</em> 2022.06.07.495102 (2022) doi:<a href="https://doi.org/10.1101/2022.06.07.495102">10.1101/2022.06.07.495102</a>.</div>
</div>
<div id="ref-Rueden2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">92. </div><div class="csl-right-inline">Rueden, C. T. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pbio.3000340">Scientific community
image forum: A discussion forum for scientific image software</a>.
<em>PLoS Biol.</em> <strong>17</strong>, e3000340 (2019).</div>
</div>
<div id="ref-Gomez-de-Mariscal2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">93. </div><div class="csl-right-inline">Gómez-de-Mariscal, E. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-021-01262-9"><span>DeepImageJ</span>:
A user-friendly environment to run deep learning models in
<span>ImageJ</span></a>. <em>Nat. Methods</em> <strong>18</strong>,
1192–1195 (2021).</div>
</div>
<div id="ref-Ouyang2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">94. </div><div class="csl-right-inline">Ouyang, W., Mueller, F., Hjelmare, M.,
Lundberg, E. &amp; Zimmer, C. <a href="https://doi.org/10.1038/s41592-019-0627-0"><span>ImJoy</span>: An
open-source computational platform for the deep learning era</a>.
<em>Nat. Methods</em> <strong>16</strong>, 1199–1200 (2019).</div>
</div>
<div id="ref-Shah" class="csl-entry" role="listitem">
<div class="csl-left-margin">95. </div><div class="csl-right-inline">Shah, R., Gogoberidze, N. &amp; Cimini, B. <a href="https://github.com/bilayer-containers/bilayers">Bilayers</a>.</div>
</div>
<div id="ref-Hidalgo-Cenalmor2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">96. </div><div class="csl-right-inline">Hidalgo-Cenalmor, I. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-024-02295-6"><span>DL4MicEverywhere</span>:
Deep learning for microscopy made flexible, shareable and
reproducible</a>. <em>Nat. Methods</em> <strong>21</strong>, 925–927
(2024).</div>
</div>
<div id="ref-jupyter" class="csl-entry" role="listitem">
<div class="csl-left-margin">97. </div><div class="csl-right-inline">Kluyver, T. <em>et al.</em> <a href="https://eprints.soton.ac.uk/403913/">Jupyter notebooks - a
publishing format for reproducible computational workflows</a>. in
<em>Positioning and power in academic publishing: Players, agents and
agendas</em> (eds. Loizides, F. &amp; Scmidt, B.) 87–90 (IOS Press,
Netherlands, 2016).</div>
</div>
<div id="ref-Kreshuk2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">98. </div><div class="csl-right-inline">Kreshuk, A. &amp; Zhang, C. <a href="https://link.springer.com/protocol/10.1007/978-1-4939-9686-5_21">Machine
learning: Advanced image segmentation using ilastik</a>. <em>Methods
Mol. Biol.</em> <strong>2040</strong>, 449–463 (2019).</div>
</div>
<div id="ref-arganda-carreras2017" class="csl-entry" role="listitem">
<div class="csl-left-margin">99. </div><div class="csl-right-inline">Arganda-Carreras, I. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btx180">Trainable
<span>Weka</span> <span>Segmentation</span>: A machine learning tool for
microscopy pixel classification</a>. <em>Bioinformatics</em>
<strong>33</strong>, 2424–2426 (2017).</div>
</div>
<div id="ref-fazeli2020" class="csl-entry" role="listitem">
<div class="csl-left-margin">100. </div><div class="csl-right-inline">Fazeli, E. <em>et al.</em> <a href="https://doi.org/10.12688/f1000research.27019.1">Automated cell
tracking using <span>StarDist</span> and <span>TrackMate</span></a>.
<em>F1000Research</em> <strong>9</strong>, 1279 (2020).</div>
</div>
<div id="ref-krizhevsky2012" class="csl-entry" role="listitem">
<div class="csl-left-margin">101. </div><div class="csl-right-inline">Krizhevsky, A., Sutskever, I. &amp; Hinton, G.
E. <a href="https://proceedings.neurips.cc/paper_files/paper/2012/hash/c399862d3b9d6b76c8436e924a68c45b-Abstract.html"><span>ImageNet</span>
<span>Classification</span> with <span>Deep</span>
<span>Convolutional</span> <span>Neural</span>
<span>Networks</span></a>. in <em>Advances in <span>Neural</span>
<span>Information</span> <span>Processing</span>
<span>Systems</span></em> vol. 25 (Curran Associates, Inc., 2012).</div>
</div>
<div id="ref-ahlers2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">102. </div><div class="csl-right-inline">Ahlers, J. <em>et al.</em> Napari: A
multi-dimensional image viewer for <span>Python</span>. (2023) doi:<a href="https://doi.org/10.5281/zenodo.8115575">10.5281/zenodo.8115575</a>.</div>
</div>
<div id="ref-russell2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">103. </div><div class="csl-right-inline">Russell, C. T. <em>et al.</em> Bia-binder:
<span>A</span> web-native cloud compute service for the bioimage
analysis community. (2024) doi:<a href="https://doi.org/10.48550/arXiv.2411.12662">10.48550/arXiv.2411.12662</a>.</div>
</div>
<div id="ref-follain2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">104. </div><div class="csl-right-inline">Follain, G. <em>et al.</em> Fast label-free
live imaging reveals key roles of flow dynamics and
<span>CD44</span>-<span>HA</span> interaction in cancer cell arrest on
endothelial monolayers. (2024) doi:<a href="https://doi.org/10.1101/2024.09.30.615654">10.1101/2024.09.30.615654</a>.</div>
</div>
<div id="ref-moen2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">105. </div><div class="csl-right-inline">Moen, E. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-019-0403-1">Deep learning for
cellular image analysis</a>. <em>Nature Methods</em>
<strong>16</strong>, 1233–1246 (2019).</div>
</div>
<div id="ref-pylvanainen2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">106. </div><div class="csl-right-inline">Pylvänäinen, J. W., Gómez-de-Mariscal, E.,
Henriques, R. &amp; Jacquemet, G. <a href="https://doi.org/10.1016/j.ceb.2023.102271">Live-cell imaging in
the deep learning era</a>. <em>Current Opinion in Cell Biology</em>
<strong>85</strong>, 102271 (2023).</div>
</div>
<div id="ref-laine2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">107. </div><div class="csl-right-inline">Laine, R. F., Arganda-Carreras, I., Henriques,
R. &amp; Jacquemet, G. <a href="https://doi.org/10.1038/s41592-021-01284-3">Avoiding a replication
crisis in deep-learning-based bioimage analysis</a>. <em>Nature
methods</em> <strong>18</strong>, 1136–1144 (2021).</div>
</div>
<div id="ref-heinrich2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">108. </div><div class="csl-right-inline">Heinrich, L. <em>et al.</em> <a href="https://doi.org/10.1038/s41586-021-03977-3">Whole-cell organelle
segmentation in volume electron microscopy</a>. <em>Nature</em>
<strong>599</strong>, 141–146 (2021).</div>
</div>
<div id="ref-liu2025" class="csl-entry" role="listitem">
<div class="csl-left-margin">109. </div><div class="csl-right-inline">Liu, B. <em>et al.</em> <a href="https://doi.org/10.1038/s41467-025-57541-y">Self-supervised
learning reveals clinically relevant histomorphological patterns for
therapeutic strategies in colon cancer</a>. <em>Nature
Communications</em> <strong>16</strong>, 2328 (2025).</div>
</div>
<div id="ref-moshkov2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">110. </div><div class="csl-right-inline">Moshkov, N. <em>et al.</em> <a href="https://doi.org/10.1038/s41467-024-45999-1">Learning
representations for image-based profiling of perturbations</a>.
<em>Nature Communications</em> <strong>15</strong>, 1594 (2024).</div>
</div>
<div id="ref-caicedo2018" class="csl-entry" role="listitem">
<div class="csl-left-margin">111. </div><div class="csl-right-inline">Caicedo, J. C., McQuin, C., Goodman, A., Singh,
S. &amp; Carpenter, A. E. Weakly <span>Supervised</span>
<span>Learning</span> of <span>Single</span>-<span>Cell</span>
<span>Feature</span> <span>Embeddings</span>. in <em>2018
<span>IEEE</span>/<span>CVF</span> <span>Conference</span> on
<span>Computer</span> <span>Vision</span> and <span>Pattern</span>
<span>Recognition</span></em> 9309–9318 (2018). doi:<a href="https://doi.org/10.1109/CVPR.2018.00970">10.1109/CVPR.2018.00970</a>.</div>
</div>
<div id="ref-li2018" class="csl-entry" role="listitem">
<div class="csl-left-margin">112. </div><div class="csl-right-inline">Li, Y. &amp; Shen, L. <a href="https://doi.org/10.1109/ACCESS.2018.2808938"><span class="nocase">cC</span>-<span>GAN</span>: <span>A</span>
<span>Robust</span> <span>Transfer</span>-<span>Learning</span>
<span>Framework</span> for <span>HEp</span>-2 <span>Specimen</span>
<span>Image</span> <span>Segmentation</span></a>. <em>IEEE Access</em>
<strong>6</strong>, 14048–14058 (2018).</div>
</div>
<div id="ref-morid2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">113. </div><div class="csl-right-inline">Morid, M. A., Borjali, A. &amp; Del Fiol, G. <a href="https://doi.org/10.1016/j.compbiomed.2020.104115">A scoping review
of transfer learning research on medical image analysis using
<span>ImageNet</span></a>. <em>Computers in Biology and Medicine</em>
<strong>128</strong>, 104115 (2021).</div>
</div>
<div id="ref-kochetov2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">114. </div><div class="csl-right-inline">Kochetov, B. <em>et al.</em> <a href="https://doi.org/10.1038/s42003-024-06714-4"><span>UNSEG</span>:
Unsupervised segmentation of cells and their nuclei in complex tissue
samples</a>. <em>Communications Biology</em> <strong>7</strong>, 1–14
(2024).</div>
</div>
<div id="ref-chen2020" class="csl-entry" role="listitem">
<div class="csl-left-margin">115. </div><div class="csl-right-inline">Chen, J. <em>et al.</em> The <span>Allen</span>
<span>Cell</span> and <span>Structure</span> <span>Segmenter</span>: A
new open source toolkit for segmenting <span>3D</span> intracellular
structures in fluorescence microscopy images. (2020) doi:<a href="https://doi.org/10.1101/491035">10.1101/491035</a>.</div>
</div>
<div id="ref-conrad2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">116. </div><div class="csl-right-inline">Conrad, R. &amp; Narayan, K. <a href="https://doi.org/10.1016/j.cels.2022.12.006">Instance segmentation
of mitochondria in electron microscopy images with a generalist deep
learning model trained on a diverse dataset</a>. <em>Cell Systems</em>
<strong>14</strong>, 58–71.e5 (2023).</div>
</div>
<div id="ref-fisch2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">117. </div><div class="csl-right-inline">Fisch, D. <em>et al.</em> <a href="https://doi.org/10.1038/s41586-024-07614-7">Molecular definition
of the endogenous <span>Toll</span>-like receptor signalling
pathways</a>. <em>Nature</em> <strong>631</strong>, 635–644
(2024).</div>
</div>
<div id="ref-bejarano2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">118. </div><div class="csl-right-inline">Bejarano, L. <em>et al.</em> <a href="https://doi.org/10.1016/j.ccell.2023.12.018">Interrogation of
endothelial and mural cells in brain metastasis reveals key
immune-regulatory mechanisms</a>. <em>Cancer Cell</em>
<strong>42</strong>, 378–395.e10 (2024).</div>
</div>
<div id="ref-rangel_dacosta2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">119. </div><div class="csl-right-inline">Rangel DaCosta, L., Sytwu, K., Groschner, C. K.
&amp; Scott, M. C. <a href="https://doi.org/10.1038/s41524-024-01336-0">A robust synthetic
data generation framework for machine learning in high-resolution
transmission electron microscopy (<span>HRTEM</span>)</a>. <em>npj
Computational Materials</em> <strong>10</strong>, 1–11 (2024).</div>
</div>
<div id="ref-lin2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">120. </div><div class="csl-right-inline">Lin, B. <em>et al.</em> <a href="https://doi.org/10.1038/s41524-022-00767-x">A deep learned
nanowire segmentation model using synthetic data augmentation</a>.
<em>npj Computational Materials</em> <strong>8</strong>, 1–12
(2022).</div>
</div>
<div id="ref-shorten2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">121. </div><div class="csl-right-inline">Shorten, C. &amp; Khoshgoftaar, T. M. <a href="https://doi.org/10.1186/s40537-019-0197-0">A survey on
<span>Image</span> <span>Data</span> <span>Augmentation</span> for
<span>Deep</span> <span>Learning</span></a>. <em>Journal of Big
Data</em> <strong>6</strong>, 60 (2019).</div>
</div>
<div id="ref-lecun1998" class="csl-entry" role="listitem">
<div class="csl-left-margin">122. </div><div class="csl-right-inline">Lecun, Y., Bottou, L., Bengio, Y. &amp;
Haffner, P. <a href="https://doi.org/10.1109/5.726791">Gradient-based
learning applied to document recognition</a>. <em>Proceedings of the
IEEE</em> <strong>86</strong>, 2278–2324 (1998).</div>
</div>
<div id="ref-alibrahim2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">123. </div><div class="csl-right-inline">Alibrahim, H. &amp; Ludwig, S. A.
Hyperparameter optimization: Comparing genetic algorithm against grid
search and bayesian optimization. in <em>2021 IEEE congress on
evolutionary computation (CEC)</em> 1551–1559 (2021). doi:<a href="https://doi.org/10.1109/CEC45853.2021.9504761">10.1109/CEC45853.2021.9504761</a>.</div>
</div>
<div id="ref-ilievski2017" class="csl-entry" role="listitem">
<div class="csl-left-margin">124. </div><div class="csl-right-inline">Ilievski, I., Akhtar, T., Feng, J. &amp;
Shoemaker, C. <a href="https://doi.org/10.1609/aaai.v31i1.10647">Efficient hyperparameter
optimization for deep learning algorithms using deterministic RBF
surrogates</a>. <em>Proceedings of the AAAI Conference on Artificial
Intelligence</em> <strong>31</strong>, (2017).</div>
</div>
</div>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    // Ensure there is a toggle, if there isn't float one in the top right
    if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
      const a = window.document.createElement('a');
      a.classList.add('top-right');
      a.classList.add('quarto-color-scheme-toggle');
      a.href = "";
      a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
      const i = window.document.createElement("i");
      i.classList.add('bi');
      a.appendChild(i);
      window.document.body.appendChild(a);
    }
    setColorSchemeToggle(hasAlternateSentinel())
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./glossary.html" class="pagination-link" aria-label="Glossary">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Glossary</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./6-image-restoration-appendix.html" class="pagination-link" aria-label="3D-RCAN for Image Restoration">
        <span class="nav-page-text"><span class="chapter-number">A</span>&nbsp; <span class="chapter-title">3D-RCAN for Image Restoration</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>